### PR TITLE
HL-313

### PR DESCRIPTION
**Pull request by pondpiu**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-313)

for scratched runner 
- move to bottom of the table
- strike through the label
- disable clicking

image

### CODE REVIEW

File: src/components/Races/RaceTable.tsx [hunk 1]
```diff
@@ -1,59 +1,77 @@
...
-              </tr>
-            </thead>
-            <tbody className="bg-white divide-y divide-gray-200">
-              {runners.map(runner => (
-                <RaceTableRow runner={runner} onClick={onClickRunner} />
-              ))}
-            </tbody>
-          </table>
+const isScratchedRunner = (runner: Runner) =>
+  ["LateScratched", "Scratched"].includes(runner.bettingStatus);
```
Comment (brettvniekerk) [inline src/components/Races/RaceTable.tsx:11]: could be:
```ts
runner.bettingStatus === "Scratched" || "LateScratched";
```

File: src/components/Races/RaceTable.tsx [hunk 2]
```diff
@@ -1,59 +1,77 @@
...
+
+export const RaceTable: React.FC<Props> = ({ runners, onClickRunner }) => {
+  const openBetRunners =
+    runners?.filter(runner => !isScratchedRunner(runner)) ??
```
Comment (brettvniekerk) [inline src/components/Races/RaceTable.tsx:15]: could make `isScratchedRunner` a util function; could extract `openBetRunners` to a util instead called `getValidRunners` or something along those lines

Comment (brettvniekerk) [reply src/components/Races/RaceTable.tsx:15]: `getValidRunners` could return an object:
```ts
{
    valid: ...
    scratched: ...
}
```
or something

File: src/components/Races/RaceTableRow.tsx [hunk 3]
```diff
@@ -6,31 +7,59 @@ import utils from "../../utils";
...
   const { number, name, barrier, odds, handicapWeight, last5Starts } =
     runner || {};
 
   return (
     <tr
-      className="cursor-pointer hover:bg-gray-100"
-      onClick={() => onClick(runner)}
+      className={classnames({
+        "cursor-pointer hover:bg-gray-100": !isScratched,
+        "cursor-not-allowed": isScratched
```
Comment (brettvniekerk) [inline src/components/Races/RaceTableRow.tsx:25]: `cursor-not-allowed` might look a bit ugly -- could probably just use `cursor-default`

File: src/components/Races/RaceTableRow.tsx [hunk 4]
```diff
@@ -6,31 +7,59 @@ import utils from "../../utils";
...
+      })}
+      onClick={() => {
```
Comment (brettvniekerk) [inline src/components/Races/RaceTableRow.tsx:27]: if you want:
```ts
onClick={() => !isScratched && onClick(runner)}
```

File: src/components/Races/RaceTableRow.tsx [hunk 5]
```diff
@@ -6,31 +7,59 @@ import utils from "../../utils";
...
+        if (!isScratched) {
+          onClick(runner);
+        }
+      }}
     >
-      <td className="px-1 py-4 whitespace-nowrap bg-gray-200">
+      <td className={"px-1 py-4 whitespace-nowrap bg-gray-200"}>
```
Comment (brettvniekerk) [inline src/components/Races/RaceTableRow.tsx:33]: why the change to curly braces?

File: src/types/meets.ts [hunk 6]
```diff
@@ -14,6 +14,7 @@ export type Runner = {
   proposition_id_hash: string;
   barrier: number;
   signature: EcSignature;
+  bettingStatus: "Open" | "Scratched" | "LateScratched";
```
Comment (brettvniekerk) [inline src/types/meets.ts:17]: inline with comment on BE pr -- could just be `status` rather than `bettingStatus` but no matter

**Review by brettvniekerk — COMMENTED** [review]
few things to consider

**Review by Full-Node-Capital — APPROVED** [review]
LGTM

File: src/components/Races/RaceTable.tsx (hunk 1 above)
Comment (pondpiu) [reply src/components/Races/RaceTable.tsx:11]: above syntax doesn't return boolean
we could do as below, but i don't like it :v
```
runner.bettingStatus === "Scratched" || runner.bettingStatus === "LateScratched";
```

File: src/components/Races/RaceTableRow.tsx (hunk 4 above)
Comment (pondpiu) [reply src/components/Races/RaceTableRow.tsx:27]: no, i don't want 😂

File: src/components/Races/RaceTableRow.tsx (hunk 5 above)
Comment (pondpiu) [reply src/components/Races/RaceTableRow.tsx:33]: oops, typo from classnames

File: src/components/Races/RaceTable.tsx (hunk 2 above)
Comment (pondpiu) [reply src/components/Races/RaceTable.tsx:15]: not worth it at this point IMO

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

**Review by JasminFarhi — APPROVED** [review]
LGTM
